tests/util/minikube/cmd/minikube.go

```diff
@@ -0,0 +1,174 @@
+		"apiserver.admission-control=\"NamespaceLifecycle,LimitRanger,ServiceAccount,PersistentVolumeLabel," +
+			"DefaultStorageClass,DefaultTolerationSeconds,MutatingAdmissionWebhook,ValidatingAdmissionWebhook,ResourceQuota\""},
+		`A set of key=value pairs that describe configuration that may be passed to different components.
+		The key should be '.' separated, and the first part before the dot is the component to apply the configuration to.
+		Valid components are: kubelet, apiserver, controller-manager, etcd, proxy, scheduler.`)
+	setupCmd.PersistentFlags().Uint16Var(&sa.cpus, "cpus", 4, "Number of CPUs allocated to the "+
+		"minikube VM")
+	setupCmd.PersistentFlags().Uint64Var(&sa.memory, "memory", 8192, "Amount of RAM allocated to"+
+		" the minikube VM in MB")
+	setupCmd.PersistentFlags().StringVar(&sa.kubernetesVersion, "kubernetes-version", "v1.10.0",
+		"Kubernetes version to use for setting up minikube")
+	setupCmd.PersistentFlags().StringSliceVar(&sa.insecureRegistry, "insecure-registry", []string{"localhost:5000"},
+		"Insecure Docker registries to pass to the Docker daemon.  The default service CIDR range will automatically"+
+			" be added.")
+	setupCmd.PersistentFlags().StringVar(&sa.vmDriver, "vm-driver", "none", "vm driver for minikube"+
+		" in Istio. Supported are none, hyperkit and kvm2")
+	setupCmd.PersistentFlags().Uint16Var(&sa.remoteRegistryPort, "remoteRegistryPort", 5000, "port"+
+		" number where registry can be port-forwarded on user's machine. Default is 5000, same as local registry port.")
+	setupCmd.PersistentFlags().StringVar(&sa.kubeconfig, "kubeconfig", "",
+		"Use a Kubernetes configuration file instead of in-cluster configuration")
+	return setupCmd
+}
+
+func teardownCfgCmd(rawArgs []string, printf, fatalf shared.FormatFn) *cobra.Command {
+	sa := &minikubeSetupArgs{}
+	teardownCmd := &cobra.Command{
+		Use:   "delete",
+		Short: "deletes minikube environmnet",
+		Run: func(cmd *cobra.Command, args []string) {
+			deleteMinikubeEnvironment(sa, printf, fatalf)
+		},
+	}
+	teardownCmd.PersistentFlags().StringVar(&sa.kubeconfig, "kubeconfig", "",
+		"Use a Kubernetes configuration file instead of in-cluster configuration")
+	return teardownCmd
+}
+
+// setupMinikubeEnvironment starts minikube and sets up local registry in minikube's k8's cluster.
+func setupMinikubeEnvironment(sa *minikubeSetupArgs, printf, fatalf shared.FormatFn) {
+	goPath := os.Getenv("GOPATH")
+	if len(goPath) == 0 {
+		fatalf("GOPATH not set.")
+	}
+
+	startMinikube(sa, printf, fatalf)
+	setupDockerToTalkToMinikube(printf, fatalf)
+	checkMinikubeRunning(fatalf)
+
+	printf("#Setting up in-cluster local registry in minikube kuberntes cluster#")
+	cmd.SetupLocalRegistry(sa.remoteRegistryPort, sa.kubeconfig, printf, fatalf)
+	printf("")
+	printf("#Minikube Setup is done#")
+	printf("#Please export HUB=localhost:%d in your console#", sa.remoteRegistryPort)
+}
+
+func startMinikube(sa *minikubeSetupArgs, printf, fatalf shared.FormatFn) {
```

Comment:
I really don't like hardcoded commands in go - hard to tweak. Anything wrong with calling a shell script as before ? There are many other options in minikube.  This s going to be too restrictive.

Reply:
+ @nmittler 
@costinm : Aim of this was to move away from sh scripts and provide user one simple command to setup the minikube test environment...

One way I can think is that all the args that are passed to minikube start are passed in as cmd line args...  what do you guys think?

Reply:
What you have here is what Oz and I were asking for. Our intent is to move as much code as possible into Go so that it's platform-agnostic, easily reused from other Go code, and easier to read/understand/maintain.